src/datepicker/datepicker-input.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  Directive,
+  Input,
+  ComponentRef,
+  ElementRef,
+  ViewContainerRef,
+  Renderer,
+  ComponentFactoryResolver,
+  NgZone,
+  TemplateRef,
+  forwardRef
+} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+
+import {NgbDate} from './ngb-date';
+import {NgbDatepicker} from './datepicker';
+import {DayTemplateContext} from './datepicker-day-template-context';
+import {NgbDateParserFormatter} from './ngb-date-parser-formatter';
+
+import {positionElements} from '../util/positioning';
+
+const NGB_DATEPICKER_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => NgbInputDatepicker),
+  multi: true
+};
+
+@Directive({
+  selector: 'input[ngbDatepicker]',
+  exportAs: 'ngbDatepicker',
+  host: {'(change)': 'manualDateChange($event.target.value)', '(keyup.esc)': 'close()'},
+  providers: [NGB_DATEPICKER_VALUE_ACCESSOR]
+})
+export class NgbInputDatepicker implements ControlValueAccessor {
+  private _cRef: ComponentRef<NgbDatepicker> = null;
+  private _model: NgbDate;
+  private _zoneSubscription: any;
+
+  /**
+   * Reference for the custom template for the day display
+   */
+  @Input() dayTemplate: TemplateRef<DayTemplateContext>;
+
+  /**
+   * First day of the week, 0=Sun, 1=Mon, etc.
+   */
+  @Input() firstDayOfWeek: number;
+
+  /**
+   * Callback to mark a given date as disabled
+   */
+  @Input() markDisabled: (date: {year: number, month: number, day: number}) => boolean;
+
+  /**
+   * Min date for the navigation. If not provided will be 10 years before today or `startDate`
+   */
+  @Input() minDate: {year: number, month: number, day: number};
+
+  /**
+   * Max date for the navigation. If not provided will be 10 years from today or `startDate`
+   */
+  @Input() maxDate: {year: number, month: number, day: number};
+
+  /**
+   * Whether to display navigation or not
+   */
+  @Input() showNavigation: boolean;
+
+  /**
+   * Whether to display days of the week
+   */
+  @Input() showWeekdays: boolean;
+
+  /**
+   * Whether to display week numbers
+   */
+  @Input() showWeekNumbers: boolean;
+
+  /**
+   * Date to open calendar with. If nothing provided, calendar will open with current month.
+   * Use 'navigateTo(date)' as an alternative
+   */
+  @Input() startDate: {year: number, month: number};
```

Comment:
Once again, this is not really part of this PR. If this needs to be discussed please create a separate issue.